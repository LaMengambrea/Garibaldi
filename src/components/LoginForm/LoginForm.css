.LoginForm {
    color: #828282;
    padding: 0px 20px 0px 20px;
    height: 100vh;
}

.marginTop{
    margin-top: 20px;
}

#forgotPasswordBtn{
    color: #828282;
    cursor: pointer;
}

#forgotPasswordBtn:hover{
    font-weight: bold;
}