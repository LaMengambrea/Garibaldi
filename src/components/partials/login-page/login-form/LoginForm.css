.LoginForm {
    color: #828282;
    padding: 0px 20px 0px 20px;
}

.marginTop{
    margin-top: 20px;
}

#forgotPasswordBtn{
    color: #828282;
    cursor: pointer;
}

#forgotPasswordBtn:hover{
    font-weight: bold;
}

#mainLogo{
    content: url("../../../../content/images/gray_logo.png");
    width: 300px;
    margin: 25px 0px;
}