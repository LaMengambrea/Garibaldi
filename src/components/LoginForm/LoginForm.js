import React, {Component} from 'react';
import './LoginForm.css';
import '../../Main.css';

export default class LoginForm extends Component {
  render() {
    return (
      <div className="LoginForm">
          <div className="Row">
              <div className="col-xs-0 col-md-4"></div>
              <div className="col-xs-12 col-md-4">
                  <div className="row">
                      Aqui va el logo
                  </div>
                  <div className="row marginTop">
                      <center>Entra ya y empieza a descubrir arte, artistas y galerías.</center>
                  </div>
                  <div className="row marginTop">
                      <div className="form-group">
                        <input type="text" placeholder="Nombre de usuario" className="form-control" id="username"/>
                        <input type="text" placeholder="Contraseña" className="form-control" id="password"/>
                        <center>
                            <div className="checkbox">
                                <label><input type="checkbox" value=""/>Recuérdame</label>
                            </div>
                        </center>
                      </div>
                  </div>
                  <div className="row divider marginTop"></div>
                  <div className="row marginTop">
                      <center><a id="forgotPasswordBtn">¿Olvidaste tu nombre de usuario o contraseña?</a></center>
                  </div>
              </div>
              <div className="col-xs-0 col-md-4"></div>
          </div>
      </div>
    );
  }
}