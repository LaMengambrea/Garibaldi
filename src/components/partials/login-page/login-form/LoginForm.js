import React, {Component, PropTypes} from 'react'
import {Link} from 'react-router';
import {connect} from 'react-redux'
import {validateObligatoryFields, getFieldValue, getFieldIndex} from '../../../../utils/fieldValidations'
import DividerComponent from '../../../ui/divider/DividerComponent'
import DefaultButton from '../../../ui/buttons/DefaultButton'
import InputFieldComponent from '../../../ui/input-field/InputFieldComponent'
import * as constants from '../../../../redux/constants'
import '../../../../Main.css'
import './LoginForm.css'
import axios from 'axios'
import {getForm, FormType} from '../../../../utils/forms/formUtils'
import {NotificationTypes} from '../../../alerts/notifications/NotificationTypes'

class LoginForm extends Component {
    constructor(props)
    {
        super(props)

        this.state = {
            inputFields: getForm(FormType.LOGIN)
        }
    }

    handleOnChange(event, index, value){
        let inputFieldsCopy = [...this.state.inputFields]
        let currentFieldIndex = getFieldIndex(inputFieldsCopy, event.target.id)
        inputFieldsCopy[currentFieldIndex].defaultValue = event.target.value

        this.setState({inputFields: inputFieldsCopy})
    }

    handleOnClick(event, {addNotification, clearAllNotifications, receiveCurrentUser}){
        clearAllNotifications();
        let inputFieldsCopy = [...this.state.inputFields]
        let result = validateObligatoryFields(this.state.inputFields)

        if(result.valid){
            let usernameValue = getFieldValue(inputFieldsCopy, "username").defaultValue
            let passwordValue = getFieldValue(inputFieldsCopy, "password").defaultValue
            
            // this.props.clearAllNotifications()
            // axios.get('api/mosaic')
            // .then(function (response) {
            //     sessionStorage.setItem('currentUser', response.data);
            // })
            // .catch(function (error) {
            //     this.props.addNotification({type: NotificationTypes.DANGER, contentType: "text", message: error});
            // })
            sessionStorage.setItem('currentUser', "test");
            receiveCurrentUser({test: "test"})
            window.location = './home'
        } else {
            addNotification({type: NotificationTypes.DANGER, contentType: "text", message: "Ingrese la información de los campos marcados en rojo"})
        }
        this.setState({inputFields: result.fieldList})
    }

    render() {
        return (
        <div className="LoginForm row">
            <div className="col-xs-0 col-md-4"></div>
            <div className="col-xs-12 col-md-4">
                <div className="row marginTop">
                    <center>
                        <img src="" id="mainLogo" alt=""/>
                    </center>
                </div>
                <div className="row marginTop">
                    <center>Entra ya y empieza a descubrir arte, artistas y galerías.</center>
                </div>
                <div className="row marginTop">
                    <div className="form-group">
                        {
                                this.state.inputFields.map((item, key) => <InputFieldComponent key={key}
                                                                        inputType={item.inputType} 
                                                                        hintText={item.hintText}
                                                                        floatingLabelText={item.floatingLabelText}
                                                                        className={item.className}
                                                                        id={item.id}
                                                                        type={item.type}
                                                                        errorText={item.errorText}
                                                                        options={item.options}
                                                                        defaultValue={item.defaultValue}
                                                                        onChange={event => this.handleOnChange(event)}/>)
                                }
                        <center>
                            <DefaultButton
                                label="Iniciar Sesión"
                                labelPosition="after"
                                floatStyle="center"
                                onTouchTap={event => this.handleOnClick(event, this.props)}
                                className="marginTop"
                                />
                        </center>
                    </div>
                </div>
                <div className="row marginTop">
                    <DividerComponent />
                </div>
                <div className="row marginTop marginBottom">
                    <center><Link to="/forgotPassword" id="forgotPasswordBtn">¿Olvidaste tu nombre de usuario o contraseña?</Link></center>
                </div>
            </div>
            <div className="col-xs-0 col-md-4"></div>
        </div>
        );
  }
}

LoginForm.displayName = 'LoginForm'

LoginForm.propTypes = {
  addNotification: PropTypes.func,
  clearAllNotifications: PropTypes.func,
  receiveCurrentUser: PropTypes.func
}

export const mapDispatchToProps = dispatch => ({
  addNotification: notification => dispatch({type: constants.ADD_NOTIFICATION, notification}),
  clearAllNotifications: () => dispatch({type: constants.CLEAR_ALL_NOTIFICATIONS}),
  receiveCurrentUser: user => dispatch({type: constants.CURRENT_USER_RECIEVED, user})
})

export default connect(null, mapDispatchToProps)(LoginForm)