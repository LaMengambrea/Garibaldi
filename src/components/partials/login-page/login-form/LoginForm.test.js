import LoginForm from './LoginForm';
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

let store = {
    subscribe: () => {},
    dispatch: () => {},
    getState: () => {
      return {
          addNotification: jest.fn(),
          clearAllNotifications: jest.fn(),
          receiveCurrentUser: jest.fn(),
          loading: jest.fn()
      };
    }
}

it('LoginForm renders', () => {
  const tree = renderer.create(
    <Provider store={store}>
        <MuiThemeProvider>
            <LoginForm/>
        </MuiThemeProvider>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});