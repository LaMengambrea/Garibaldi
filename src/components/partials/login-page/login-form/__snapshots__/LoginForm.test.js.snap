exports[`test LoginForm renders 1`] = `
<div
  className="LoginForm row">
  <div
    className="col-xs-0 col-md-4" />
  <div
    className="col-xs-12 col-md-4">
    <div
      className="row marginTop">
      <center>
        <img
          alt=""
          id="mainLogo"
          src="" />
      </center>
    </div>
    <div
      className="row marginTop">
      <center>
        Entra ya y empieza a descubrir arte, artistas y galerías.
      </center>
    </div>
    <div
      className="row marginTop">
      <div
        className="form-group">
        <div
          className="obligatoryField TextField"
          style={
            Object {
              "backgroundColor": "transparent",
              "cursor": "auto",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": 16,
              "height": 72,
              "lineHeight": "24px",
              "muiPrepared": true,
              "position": "relative",
              "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": "100%",
            }
          }>
          <label
            className={undefined}
            htmlFor="username"
            onTouchTap={undefined}
            style={
              Object {
                "color": "#9e9e9e",
                "lineHeight": "22px",
                "muiPrepared": true,
                "pointerEvents": "none",
                "position": "absolute",
                "top": 38,
                "transform": "scale(1) translate(0, 0)",
                "transformOrigin": "left top",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "zIndex": 1,
              }
            }>
            Nombre de usuario
          </label>
          <div
            style={
              Object {
                "bottom": 12,
                "color": "rgba(0, 0, 0, 0.3)",
                "muiPrepared": true,
                "opacity": 0,
                "position": "absolute",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              }
            }>
            Ingresa el nombre de usuario
          </div>
          <input
            defaultValue=""
            disabled={false}
            id="username"
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                "appearance": "textfield",
                "backgroundColor": "rgba(0,0,0,0)",
                "border": "none",
                "boxSizing": "border-box",
                "color": "#757575",
                "cursor": "inherit",
                "font": "inherit",
                "height": "100%",
                "marginTop": 14,
                "muiPrepared": true,
                "outline": "none",
                "padding": 0,
                "position": "relative",
                "width": "100%",
              }
            }
            type="text" />
          <div>
            <hr
              aria-hidden="true"
              style={
                Object {
                  "borderBottom": "solid 1px",
                  "borderColor": "#9e9e9e",
                  "borderLeft": "none",
                  "borderRight": "none",
                  "borderTop": "none",
                  "bottom": 8,
                  "boxSizing": "content-box",
                  "margin": 0,
                  "muiPrepared": true,
                  "position": "absolute",
                  "width": "100%",
                }
              } />
            <hr
              aria-hidden="true"
              style={
                Object {
                  "borderBottom": "solid 2px",
                  "borderColor": "#757575",
                  "borderLeft": "none",
                  "borderRight": "none",
                  "borderTop": "none",
                  "bottom": 8,
                  "boxSizing": "content-box",
                  "margin": 0,
                  "muiPrepared": true,
                  "position": "absolute",
                  "transform": "scaleX(0)",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "width": "100%",
                }
              } />
          </div>
          
        </div>
        <div
          className="obligatoryField TextField"
          style={
            Object {
              "backgroundColor": "transparent",
              "cursor": "auto",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": 16,
              "height": 72,
              "lineHeight": "24px",
              "muiPrepared": true,
              "position": "relative",
              "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": "100%",
            }
          }>
          <label
            className={undefined}
            htmlFor="password"
            onTouchTap={undefined}
            style={
              Object {
                "color": "#9e9e9e",
                "lineHeight": "22px",
                "muiPrepared": true,
                "pointerEvents": "none",
                "position": "absolute",
                "top": 38,
                "transform": "scale(1) translate(0, 0)",
                "transformOrigin": "left top",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "zIndex": 1,
              }
            }>
            Ingresa la contraseña
          </label>
          <div
            style={
              Object {
                "bottom": 12,
                "color": "rgba(0, 0, 0, 0.3)",
                "muiPrepared": true,
                "opacity": 0,
                "position": "absolute",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              }
            }>
            Contraseña
          </div>
          <input
            defaultValue=""
            disabled={false}
            id="password"
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                "appearance": "textfield",
                "backgroundColor": "rgba(0,0,0,0)",
                "border": "none",
                "boxSizing": "border-box",
                "color": "#757575",
                "cursor": "inherit",
                "font": "inherit",
                "height": "100%",
                "marginTop": 14,
                "muiPrepared": true,
                "outline": "none",
                "padding": 0,
                "position": "relative",
                "width": "100%",
              }
            }
            type="password" />
          <div>
            <hr
              aria-hidden="true"
              style={
                Object {
                  "borderBottom": "solid 1px",
                  "borderColor": "#9e9e9e",
                  "borderLeft": "none",
                  "borderRight": "none",
                  "borderTop": "none",
                  "bottom": 8,
                  "boxSizing": "content-box",
                  "margin": 0,
                  "muiPrepared": true,
                  "position": "absolute",
                  "width": "100%",
                }
              } />
            <hr
              aria-hidden="true"
              style={
                Object {
                  "borderBottom": "solid 2px",
                  "borderColor": "#757575",
                  "borderLeft": "none",
                  "borderRight": "none",
                  "borderTop": "none",
                  "bottom": 8,
                  "boxSizing": "content-box",
                  "margin": 0,
                  "muiPrepared": true,
                  "position": "absolute",
                  "transform": "scaleX(0)",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "width": "100%",
                }
              } />
          </div>
          
        </div>
        <center>
          <div
            className="marginTop"
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                "backgroundColor": "#ffffff",
                "borderRadius": 2,
                "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                       0 1px 4px rgba(0, 0, 0, 0.12)",
                "boxSizing": "border-box",
                "color": "rgba(0, 0, 0, 0.87)",
                "display": "inline-block",
                "float": "center",
                "fontFamily": "Roboto, sans-serif",
                "margin": 12,
                "minWidth": 88,
                "muiPrepared": true,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              }
            }>
            <button
              disabled={false}
              href={undefined}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              onTouchTap={[Function]}
              style={
                Object {
                  "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                  "backgroundColor": "#ffffff",
                  "border": 10,
                  "borderRadius": 2,
                  "boxSizing": "border-box",
                  "cursor": "pointer",
                  "display": "inline-block",
                  "fontFamily": "Roboto, sans-serif",
                  "fontSize": "inherit",
                  "fontWeight": "inherit",
                  "height": 36,
                  "lineHeight": "36px",
                  "margin": 0,
                  "muiPrepared": true,
                  "outline": "none",
                  "padding": 0,
                  "position": "relative",
                  "textAlign": "center",
                  "textDecoration": "none",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "verticalAlign": null,
                  "width": "100%",
                  "zIndex": 1,
                }
              }
              tabIndex={0}
              type="button">
              <div
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}>
                <div
                  style={
                    Object {
                      "backgroundColor": false,
                      "borderRadius": 2,
                      "height": 36,
                      "muiPrepared": true,
                      "top": 0,
                      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    }
                  }>
                  <span
                    style={
                      Object {
                        "color": "rgba(0, 0, 0, 0.87)",
                        "fontSize": 14,
                        "fontWeight": 500,
                        "letterSpacing": 0,
                        "margin": 0,
                        "muiPrepared": true,
                        "opacity": 1,
                        "paddingLeft": 16,
                        "paddingRight": 16,
                        "position": "relative",
                        "textTransform": "uppercase",
                        "userSelect": "none",
                      }
                    }>
                    Iniciar Sesión
                  </span>
                </div>
              </div>
            </button>
          </div>
        </center>
      </div>
    </div>
    <div
      className="row marginTop">
      <hr
        style={
          Object {
            "backgroundColor": "#9e9e9e",
            "border": "none",
            "height": 1,
            "margin": 0,
            "marginLeft": 0,
            "marginTop": -1,
            "muiPrepared": true,
          }
        } />
    </div>
    <div
      className="row marginTop marginBottom">
      <center>
        <a
          id="forgotPasswordBtn"
          onClick={[Function]}
          style={Object {}}>
          ¿Olvidaste tu contraseña?
        </a>
      </center>
    </div>
  </div>
  <div
    className="col-xs-0 col-md-4" />
</div>
`;
